### PR TITLE
added solution for task with matrix operation

### PR DESCRIPTION
**Pull request by kozhukhovsky**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
+        return mainDiagonal;
     }
 
+    /**
+     * Method return min value of matrix.
+     * @return min value of matrix.
+     */
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = Integer.MAX_VALUE;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: why we use Integer.MAX_VALUE here?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -1,19 +1,52 @@
...
+        return min;
     }
 
+    /**
+     * Method return max value of matrix.
+     * @return max value of matrix.
+     */
     public int getMax(int[][] matrix) {
-        return 0;
+        int max = Integer.MIN_VALUE;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: the same

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (kozhukhovsky) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Why not? We can use first element of this matrix, but use maximum value for this type its correct

Comment (kozhukhovsky) [reply src/main/java/core/basesyntax/MatrixOperation.java]: fixed

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (kozhukhovsky) [reply src/main/java/core/basesyntax/MatrixOperation.java]: fixed

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -1,19 +1,65 @@
...
     }
 
+    /**
+     * Returns min value of matrix.
+     *
+     * @return min value of matrix, or 0 if matrix empty
+     */
     public int getMin(int[][] matrix) {
-        return 0;
+        int min;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: we can simplify:
        if (matrix == null) {
            return 0;
        } 
int min = matrix[0][0];else {
    ...

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 4]
```diff
@@ -1,19 +1,65 @@
...
+     * Returns min value of matrix.
+     *
+     * @return min value of matrix, or 0 if matrix empty
+     */
     public int getMin(int[][] matrix) {
-        return 0;
+        if (matrix == null) {
+            return 0;
+        }
+
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MatrixOperation.java]: next time remove redundant lines

**Pull request closed without merging**